server/src/infra/repositories/access.repository.ts

```diff
@@ -155,9 +157,9 @@ export class AccessRepository implements IAccessRepository {
         )
         .andWhere(
           new Brackets((qb) => {
-            qb.where('asset.id IN (:...assetIds)', { assetIds: [...assetIds] })
+            qb.where('asset.id IN (SELECT id FROM "assetIds")')
               // still part of a live photo is in an album
-              .orWhere('asset.livePhotoVideoId IN (:...assetIds)', { assetIds: [...assetIds] });
+              .orWhere('asset.livePhotoVideoId IN (SELECT id FROM "assetIds")');
```

Comment:
This is a slow query. You can change this to `array["asset"."id", "asset"."livePhotoVideoId"] && array[:...assetIds]` and remove the CTE. This checks if there's any overlap between the two arrays.

Reply:
Thanks for pointing that out! I wasn't aware of array overlapping. Just changed it to this approach.